src/Raven.Server/Documents/Sharding/Documents/ShardedSubscriptionContext.cs

```diff
@@ -0,0 +1,183 @@
+
+                var subscription = GetShardedSubscriptionFromServerStore(_serverStore, serverStoreContext, _shardedContext, name);
+
+                if (subscription.Disabled)
+                    throw new SubscriptionClosedException($"The subscription with id '{id}' and name '{name}' is disabled and cannot be used until enabled");
+
+                return subscription;
+            }
+        }
+
+        public SubscriptionGeneralDataAndStats GetRunningSubscription(TransactionOperationContext context, long? id, string name, bool history)
+        {
+            SubscriptionGeneralDataAndStats subscription;
+            if (string.IsNullOrEmpty(name) == false)
+            {
+                subscription = GetShardedSubscriptionFromServerStore(_serverStore, context, _shardedContext, name);
+            }
+            else if (id.HasValue)
+            {
+                name = GetSubscriptionNameById(context, _shardedContext.GetShardedDatabaseName(), id.Value);
+                subscription = GetShardedSubscriptionFromServerStore(_serverStore, context, _shardedContext, name);
+            }
+            else
+            {
+                throw new ArgumentNullException("Must receive either subscription id or subscription name in order to provide subscription data");
+            }
+
+            if (_subscriptionConnectionStates.TryGetValue(subscription.SubscriptionId, out SubscriptionConnectionState subscriptionConnectionState) == false)
+                return null;
+
+            if (subscriptionConnectionState.Connection == null)
+                return null;
+
+            GetRunningSubscriptionInternal(history, subscription, subscriptionConnectionState);
+            return subscription;
+        }
+
+        public static SubscriptionGeneralDataAndStats GetShardedSubscriptionFromServerStore(ServerStore serverStore, TransactionOperationContext context, ShardedContext shardedContext, string name)
+        {
+            using BlittableJsonReaderObject subscriptionBlittable = serverStore.Cluster.Read(context, SubscriptionState.GenerateSubscriptionItemKeyName(shardedContext.DatabaseName, name));
+
+            if (subscriptionBlittable == null)
+                ThrowNotFoundException(name);
+
+            var subscriptionState = JsonDeserializationClient.SubscriptionState(subscriptionBlittable);
+            var subscriptionJsonValue = new SubscriptionGeneralDataAndStats(subscriptionState);
+            return subscriptionJsonValue;
+        }
+
+        private static void ThrowNotFoundException(string name)
+        {
+            throw new SubscriptionDoesNotExistException($"Subscription with name '{name}' was not found in server store");
+        }
+
+        public List<string> GetResponsibleNodes(TransactionOperationContext context, string name)
```

Comment:
This is not used

Reply:
its used in the `ShardedSubscriptionsHandler`
https://github.com/ravendb/ravendb/pull/12543/files#diff-778d2e60f2cc32160119a379c800500d7ed41562794c464888d47d57009dcc18R166

we wait for all repsonsible nodes to receive the index